adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -256,156 +250,157 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
+    filePath: String,
+    predicate: Option[Class[U]] = None,
+    projection: Option[Schema] = None): RDD[NucleotideContigFragment] = {
+    adamLoad[NucleotideContigFragment, U](filePath, predicate, projection)
   }
 
-  private def maybeLoadBED(filePath: String): Option[RDD[Feature]] = {
-    if (filePath.endsWith(".bed")) {
-      Some(sc.textFile(filePath).flatMap(new BEDParser().parse))
-    } else {
-      None
-    }
+  def loadGTF(filePath: String): RDD[Feature] = {
+    val records = sc.textFile(filePath).flatMap(new GTFParser().parse)
+    if (Metrics.isRecording) records.instrument() else records
   }
 
-  private def maybeLoadNarrowPeak(filePath: String): Option[RDD[Feature]] = {
-    if (filePath.toLowerCase.endsWith(".narrowpeak")) {
-      Some(sc.textFile(filePath).flatMap(new NarrowPeakParser().parse))
-    } else {
-      None
-    }
+  def loadBED(filePath: String): RDD[Feature] = {
+    val records = sc.textFile(filePath).flatMap(new BEDParser().parse)
+    if (Metrics.isRecording) records.instrument() else records
   }
 
-  private def maybeLoadVcfAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+  def loadNarrowPeak(filePath: String): RDD[Feature] = {
+    val records = sc.textFile(filePath).flatMap(new NarrowPeakParser().parse)
+    if (Metrics.isRecording) records.instrument() else records
+  }
+
+  def loadParquetFeatures[U <: ADAMPredicate[Feature]](
     filePath: String,
     predicate: Option[Class[U]] = None,
-    projection: Option[Schema] = None,
-    sd: Option[SequenceDictionary]): Option[RDD[DatabaseVariantAnnotation]] = {
-    if (filePath.endsWith(".vcf")) {
-      log.info("Reading VCF file from %s".format(filePath))
-      if (projection.isDefined) {
-        log.warn("Projection is ignored when loading a VCF file")
-      }
+    projection: Option[Schema] = None): RDD[Feature] = {
+    adamLoad[Feature, U](filePath, predicate, projection)
+  }
 
-      val job = HadoopUtil.newJob(sc)
-      val vcc = new VariantContextConverter(sd)
-      val records = sc.newAPIHadoopFile(
-        filePath,
-        classOf[VCFInputFormat], classOf[LongWritable], classOf[VariantContextWritable],
-        ContextUtil.getConfiguration(job))
+  def loadVcfAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+    filePath: String,
+    predicate: Option[Class[U]] = None,
+    sd: Option[SequenceDictionary] = None): RDD[DatabaseVariantAnnotation] = {
 
-      Some(applyPredicate(records.map(p => vcc.convertToAnnotation(p._2.get)), predicate))
-    } else
-      None
+    val job = HadoopUtil.newJob(sc)
+    val vcc = new VariantContextConverter(sd)
+    val records = sc.newAPIHadoopFile(
+      filePath,
+      classOf[VCFInputFormat], classOf[LongWritable], classOf[VariantContextWritable],
+      ContextUtil.getConfiguration(job))
+    if (Metrics.isRecording) records.instrument() else records
+
+    applyPredicate(records.map(p => vcc.convertToAnnotation(p._2.get)), predicate)
```

Comment:
`applyPredicate` is a convenience function to avoid needing to `RDD.filter()` over and over again.